enhancements/baremetal/ztp-metal3.md

```diff
@@ -0,0 +1,325 @@
+responsible for interactions with baremetal-operator) by applying the following
+change to the `BareMetalHost`'s spec:
+
+```yaml
+Image:
+  URL: url/of/assisted/agent.iso
+  DiskFormat: live-iso
+```
+
+After this enhancement, BMAC will instead apply:
+
+```yaml
+CustomDeploy:
+  Method: start_assisted_install
+```
+
+This instructs Ironic to find and execute a *custom deploy step* called
+`start_assisted_install`, which will be provided by an OpenShift specific
+Ironic agent plugin. This step will use D-Bus to order systemd to start the
+assisted agent and watch its status.
+
+If the assisted agent exits successfully, the Ironic agent will report success
+to Ironic and baremetal-operator. Otherwise, a failure will be recorded.
+
+The `CustomDeploy` field can be set from the very beginning as well. Then BMO
+will conduct inspection, apply firmware settings (if any) and prepare for
+the installation immediately.
+
+### Ironic changes
+
+Two small tweaks will be required in Ironic:
+
+- Allow deploy steps without a timeout, so that Ironic does not time out
+  the `start_assisted_install` deploy steps.
+
+- Allow accepting additional kernel parameters instead of overriding the
+  parameters completely.
+
+### Risks and Mitigations
+
+- More containers will be downloaded on the ramdisk start-up. We'll need to
+  account for more RAM and a longer start-up time.
+
+- Users will be responsible for correctly linking `BareMetalHost`s with
+  `InfraEnv` resources.
+
+## Design Details
+
+### Open Questions
+
+- The assisted mode of operation is to boot the assisted agent and wait for
+  instructions from the service (and indirectly - from the user). During this
+  time, the `BareMetalHost` state will be `provisioning`, and timeouts may
+  apply. We may need a way to tell Ironic to never time out this operation,
+  e.g. via custom timeouts per deploy step.
+
+### Test Plan
+
+- Building the new ISO will be covered by the existing ZTP e2e tests.
+- A new ZTP job will need to be created for network booting.
+
+### Graduation Criteria
+
+This work will be GA immediately since there is no phasing possible or planned.
+Only ZTP will be affected. The new behavior will be fully opt-in from the
+assisted service's (more specifically, BMAC) point of view.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+This change does not introduce new resources, so upgrading or downgrading them
+is not a concern.
+
+For the new flow to apply, it must be explicitly requested through the
+`BareMetalHost` API (by setting `CustomDeploy` instead of `Image`). Otherwise,
+the old flow is used.
+
+There are no plans to remove the code that supports the *live ISO* flow from
+Ironic, Metal3 or OpenShift.
+
+### Version Skew Strategy
+
+The baremetal-operator will need to be updated before the new flow will be
```

Comment:
This is a prerequisite only for PXE or it's relevant for the live-ISO as well?
In other words, can the assisted-service set the `start_assisted_install` deploy method and update the PreprovisioningImage status with an image containing both agents  on HUB cluster in version 4.10 (or older)

Reply:
> set the start_assisted_install deploy method 

This deploy method will only be added in 4.11, so no.